controller.go

```diff
@@ -234,3 +267,75 @@ func ensureNodeExists(instances cloudprovider.Instances, node *v1.Node) (bool, e
 
 	return true, nil
 }
+
+// If clusterAPI and clusterID are set, then we are running
+// inside Giantnetes, so try to get certificates and create
+// kubernetes client for guest cluster. If not proceed with
+// kubeconfig file.
+func newGuestClient(clusterID, clusterAPI, master, kubeconfig string) (clientset.Interface, error) {
+	if clusterAPI != "" && clusterID != "" {
+		return newGuestClientFromSecret(clusterAPI, clusterID)
+	}
```

Comment:
I'm unclear. Where is the controller supposed to run? In guest or host cluster? I thought only in host. If not, why is this flexible?

Reply:
You're right, it should run only in host cluster. So i removed this unecessary logic. See updated change.